src/material/icon/icon-registry.ts

```diff
@@ -426,38 +423,22 @@ export class MatIconRegistry implements OnDestroy {
    * from it.
    */
   private _loadSvgIconFromConfig(config: SvgIconConfig): Observable<SVGElement> {
-    return this._fetchUrl(config.url)
-        .pipe(map(svgText => this._createSvgElementForSingleIcon(svgText, config.options)));
+    return this._fetchUrl(config.url).pipe(
+      tap(svgText => config.svgText = svgText),
+      map(() => this._svgElementFromConfig(config as LoadedSvgIconConfig))
+    );
   }
 
   /**
-   * Loads the content of the icon set URL specified in the SvgIconConfig and creates an SVG element
-   * from it.
+   * Loads the content of the icon set URL specified in the
+   * SvgIconConfig and attaches it to the config.
    */
-  private _loadSvgIconSetFromConfig(config: SvgIconConfig): Observable<SVGElement> {
-    // If the SVG for this icon set has already been parsed, do nothing.
-    if (config.svgElement) {
-      return observableOf(config.svgElement);
+  private _loadSvgIconSetFromConfig(config: SvgIconConfig): Observable<null> {
+    if (config.svgText) {
+      return observableOf(null);
     }
 
-    return this._fetchUrl(config.url).pipe(map(svgText => {
-      // It is possible that the icon set was parsed and cached by an earlier request, so parsing
-      // only needs to occur if the cache is yet unset.
-      if (!config.svgElement) {
-        config.svgElement = this._svgElementFromString(svgText);
-      }
-
-      return config.svgElement;
-    }));
-  }
-
-  /**
-   * Creates a DOM element from the given SVG string, and adds default attributes.
-   */
-  private _createSvgElementForSingleIcon(responseText: string, options?: IconOptions): SVGElement {
-    const svg = this._svgElementFromString(responseText);
-    this._setSvgAttributes(svg, options);
-    return svg;
+    return this._fetchUrl(config.url).pipe(tap(svgText => config.svgText = svgText), mapTo(null));
```

Comment:
Always mapping this to null seems to defeat the point of having these streams in the first place. Would it be better to just eliminate them? E.g., change this function to `void` and propagate that through

Reply:
I agree that always emitting `null` is weird, but I don't see a nicer way of expressing that we want to trigger the request to attach the SVG text to the config, but we don't care about the result.

Reply:
I pulled down the change and tried poking at it a bit to see what felt right. I think I would get of of always mapping to `null`, since it's doing extra runtime work. We can keep the stream as `string | null` even if we don't care about the value. 

I tangentially made some edits to this code that clean up things slightly, feel free to incorporate: https://github.com/angular/components/commit/6fd70e9fbf5880f5b8e5654d8f5640dc08f7624d

Reply:
Looks like this messed up some of the error logging. I'll keep it as it is for now and I'll fix the comment about mapping it to `null`.